shell/common/surface.h

```diff
@@ -25,6 +27,11 @@ class SurfaceFrame {
                bool supports_readback,
                const SubmitCallback& submit_callback);
 
+  SurfaceFrame(sk_sp<SkSurface> surface,
+               bool supports_readback,
+               const SubmitCallback& submit_callback,
+               std::unique_ptr<GLContextResult> context_result);
```

Comment:
Added a `GLContextResult` in the constructor, so iOS platform can pass in a GLContectSwitch whose lifecycle is tied to the `SurfaceFrame`